assembly/src/parsers/blocks.rs

```diff
@@ -291,3 +301,26 @@ pub fn combine_blocks(mut blocks: Vec<CodeBlock>) -> CodeBlock {
 
     blocks.remove(0)
 }
+
+/// Returns a CodeBlock [Span] from sequence of Span blocks provided as input.
+pub fn combine_spans(spans: &mut Vec<CodeBlock>) -> CodeBlock {
+    if spans.len() == 1 {
+        return spans.remove(0);
+    }
+
+    let mut ops = Vec::<Operation>::new();
+    for block in (*spans).iter() {
```

Comment:
Could we do `for block in spans.into_iter()` here?

Reply:
Clippy is failing on `spans.into_iter()`. Shall I use `spans.into_mut()` or revert back to `(*spans).iter()`?

Reply:
`iter_mut()` is fine, if it works. You may also be able to use `.drain(0..).for_each()` and then you would not need to clear the vector at the end.

Reply:
Thanks for your suggestions. I've used `.drain(0..).for_each()` in the latest PR.